spec.html

```diff
@@ -39811,6 +40031,36 @@ <h1>
       </emu-clause>
     </emu-clause>
 
+    <emu-clause id="sec-comparetypedarrayelements" type="abstract operation">
+      <h1>
+        CompareTypedArrayElements (
+          _x_: unknown,
+          _y_: unknown,
+          _comparefn_: a function object or *undefined*,
+        ): either a normal completion containing a Number or an abrupt completion
+      </h1>
+      <dl class="header">
+      </dl>
+      <emu-alg>
+        1. Assert: _x_ is a Number and _y_ is a Number, or _x_ is a BigInt and _y_ is a BigInt.
+        1. If _comparefn_ is not *undefined*, then
+          1. Let _v_ be ? ToNumber(? Call(_comparefn_, *undefined*, « _x_, _y_ »)).
+          1. If _v_ is *NaN*, return *+0*<sub>𝔽</sub>.
+          1. Return _v_.
+        1. If _x_ and _y_ are both *NaN*, return *+0*<sub>𝔽</sub>.
+        1. If _x_ is *NaN*, return *1*<sub>𝔽</sub>.
+        1. If _y_ is *NaN*, return *-1*<sub>𝔽</sub>.
+        1. If _x_ &lt; _y_, return *-1*<sub>𝔽</sub>.
+        1. If _x_ > _y_, return *1*<sub>𝔽</sub>.
+        1. If _x_ is *-0*<sub>𝔽</sub> and _y_ is *+0*<sub>𝔽</sub>, return *-1*<sub>𝔽</sub>.
+        1. If _x_ is *+0*<sub>𝔽</sub> and _y_ is *-0*<sub>𝔽</sub>, return *1*<sub>𝔽</sub>.
+        1. Return *+0*<sub>𝔽</sub>.
```

Comment:
is there a reason this can't use `IsLessThan`, like `CompareArrayElements` does? (with a numeric rather than a string comparison)

Reply:
actually i see how the different NaN and -0/+0 behavior would make that no clearer, so i'll resolve this.